lib/vehicle_factory.rb

```diff
@@ -0,0 +1,106 @@
+class VehicleFactory
+
+    def create_vehicles(source)
+        vehicles_array = []
+        source.each do |vehicle|
+            if vehicle.keys.include?(:dol_vehicle_id)
+                vehicles_array.concat(wa_create_vehicles(source)) # .concat combining two arrays into a single array
+            elsif vehicle.keys.include?(:reg_valid_date)
+                vehicles_array.concat(ny_create_vehicles(source))
+            end
+        end
+        vehicles_array
+    end
+
+    def ny_create_vehicles(source)
+        @ny_vehicles_array = []
+        source.each do |vehicle_details|
+            vin = vehicle_details[:vin]
+            year = vehicle_details[:model_year]
+            make = vehicle_details[:make]
+            model = vehicle_details[:body_type]
+            registration_date = vehicle_details[:reg_valid_date]
+    
+            ny_vehicle = Vehicle.new({
+                vin: vin,
+                year: year,
+                make: make,
+                model: model,
+                engine: nil,
+                registration_date: registration_date,
+                plate_type: nil
+            })
+            @ny_vehicles_array << ny_vehicle
+        end
+        @ny_vehicles_array
+    end
+   
+    def wa_create_vehicles(source)
+        @wa_ev_vehicles_array = []
+        source.each do |vehicle_details|
+                vin = vehicle_details[:vin_1_10],
+                year = vehicle_details[:model_year],
+                make = vehicle_details[:make],
+                model = vehicle_details[:model]
+            
+                wa_ev_vehicle = Vehicle.new({
+                    vin: vin,
+                    year: year,
+                    make: make,
+                    model: model,
+                    engine: :ev,
+                    registration_date: nil,
+                    plate_type: nil
+                })
+            @wa_ev_vehicles_array << wa_ev_vehicle
+        end
+        @wa_ev_vehicles_array
+    end
+
+    def most_popular_make
+        hash_grouped_by_make = @wa_ev_vehicles_array.group_by do |vehicle|
+            vehicle.make
+        end
+        most_popular_make = hash_grouped_by_make.max_by do |make, vehicles_array| # .max_by finds largest array
+            vehicles_array.size
+        end
+        most_popular_make.first
```

Comment:
Awesome, love the use of enumerables here. Care with your variable names though, we dont' want to make the variable name the same as the name of the method. Can cause confusion.